### PR TITLE
Vs code flexible controllers

### PR DESCRIPTION
**Pull request by akalburgi**

### CODE REVIEW

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml [hunk 1]
```diff
@@ -46,6 +46,18 @@
         <key>name</key>
         <string>constant.language.empty-list.daml</string>
       </dict>
+      <dict>
+        <key>match</key>
+        <string>\b(choice)\b</string>
```
Comment (neil-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml]: Why the brackets here? What do they mean?

Comment (akalburgi) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml]: Nope not needed I was experimenting with something,

**Comment by akalburgi** [comment]
image

**Comment by akalburgi** [comment]
image

**Comment by akalburgi** [comment]
@shayne-fletcher-da  Yes. I just talked to Paul and Martin, to clarify the scoping. I pushed the fix for the choice keyword. But I see the controllers within a scope have a wider selection of the scope. I am going to try and fix that now

**Comment by shayne-fletcher-da** [comment]
> @shayne-fletcher-da Yes. I just talked to Paul and Martin, to clarify the scoping. I pushed the fix for the choice keyword. But I see the controllers within a scope have a wider selection of the scope. I am going to try and fix that now

Right, that's exactly what I was getting at. I think since Paul's work we now have rigorous definitions for DAML specific constructs and I anticipated they would need tinkering with/extending here. Thanks for picking this up!

**Comment by akalburgi** [comment]
> Note that pure and return aren't keywords, they are functions

Thanks for pointing that out. This file needs review and updating will grab some time with @shayne-fletcher-da Monday

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml [hunk 2]
```diff
@@ -343,6 +343,7 @@
 			| rec
       | template
       | with
+      | choice
```
Comment (shayne-fletcher-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml]: Why is there a sudden change in the indentation level from `template` onwards?

Comment (akalburgi) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage.xml]: space and tabs mixed up :(

**Pull request merged**